lib/i18n/exceptions.rb

```diff
@@ -110,4 +110,37 @@ def initialize(type, filename)
       super "can not load translations from #{filename}, the file type #{type} is not known"
     end
   end
+
+  class UnsupportedMethod < ArgumentError
+    attr_reader :method, :backend_klass
+    def initialize(method, backend_klass)
+      @method = method
+      @backend_klass = backend_klass
+      super "#{backend_klass} does not support the ##{method} method"
+    end
+  end
+
+  class InvalidFilenames < ArgumentError
+    attr_reader :invalid_files
+
+    def initialize(invalid_files)
+      @invalid_files = invalid_files
+      super <<~MSG
+        The following files are named incorrectly: #{invalid_files}.
```

Comment:
What do you think about listing the files individually?
```suggestion
        The following files are named incorrectly:
         #{invalid_files.join("\n")}

```